### ISSUE TITLE
Assertion failed while compiling oxid

### ISSUE THREAD

**Issue by dbandstra** [ziglang/zig#3661]
I'm super excited to take advantage of anonymous structs in Oxid, but something broke after the merge. Getting this when compiling oxid master:

```
Code Generation [413/2285] main...Assertion failed at /home/dbandstra/zig/src/analyze.cpp:5439 in type_has_bits. This is a bug in the Zig compiler.
```

I only know enough gdb to get this back trace:

```
#8  0x000055555608913b in type_has_bits (type_entry=0x55557581df00) at /home/dbandstra/zig/src/analyze.cpp:5439
#9  0x0000555555f9c6a6 in ir_llvm_value (g=0x55555886e810, instruction=0x55557581de30) at /home/dbandstra/zig/src/codegen.cpp:1702
#10 0x0000555555fa80f2 in ir_render_struct_field_ptr (g=0x55555886e810, executable=0x555575807140, instruction=0x555575826bd0) at /home/dbandstra/zig/src/codegen.cpp:4297
#11 0x0000555555fafe1f in ir_render_instruction (g=0x55555886e810, executable=0x555575807140, instruction=0x555575826bd0) at /home/dbandstra/zig/src/codegen.cpp:6157
#12 0x0000555555fb0752 in ir_render (g=0x55555886e810, fn_entry=0x555575807020) at /home/dbandstra/zig/src/codegen.cpp:6317
#13 0x0000555555fb6a76 in do_code_gen (g=0x55555886e810) at /home/dbandstra/zig/src/codegen.cpp:7665
#14 0x0000555555fc0f36 in codegen_build_and_link (g=0x55555886e810) at /home/dbandstra/zig/src/codegen.cpp:10181
#15 0x0000555555f9409d in main (argc=65, argv=0x7fffffffd958) at /home/dbandstra/zig/src/main.cpp:1286
```

**Comment by andrewrk** [comment]
Thanks, I'll prioritize this. This should be an easy fix.

**Comment by andrewrk** [comment]
Fixed in bf8870a60bb35062b2b25867a0f9a95f6f5397ce. I made sure Oxid builds and runs after this. Cool, multiplayer!

**Comment by dbandstra** [comment]
Thanks :) this issue was nothing but a shady marketing ploy.
